### PR TITLE
✨ Added combo functions for WETH

### PR DESCRIPTION
**Pull request by Atarpara**
## Description
Added Function `depositTo`, `depositAndApprove` and `withdrawTo`.
## Checklist

Ensure you completed **all of the steps** below before submitting your pull request:

- [x] Ran `forge fmt`?
- [x] Ran `forge snapshot`?
- [x] Ran `forge test`?

_Pull requests with an incomplete checklist will be thrown out._

### CODE REVIEW

File: src/tokens/WETH.sol [hunk 1]
```diff
@@ -84,6 +84,48 @@ contract WETH is ERC20 {
...
+    }
+
+    /// @dev Deposits `amount` ETH of the caller and mints `amount` WETH to the caller.
+    /// @dev Sets `amount` as the allowance of `spender` over the tokens of `owner`.
+    /// @notice It will update old allowance to new allowance.
+    /// Emits a {Deposit} event.
+    function depositAndApprove(address spender) external payable virtual {
+        deposit();
+        _approve(msg.sender, spender, msg.value);
+    }
```
Comment (Vectorized) [inline src/tokens/WETH.sol:107]: Not needed.

File: src/tokens/WETH.sol [hunk 2]
```diff
@@ -84,6 +84,48 @@ contract WETH is ERC20 {
...
+    /// @dev Burns `amount` WETH of the caller and sends `amount` ETH to the `to`.
+    ///
+    /// Emits a {Withdrawal} event.
+    function withdrawTo(address to, uint256 amount) public virtual {
+        _burn(msg.sender, amount);
+        /// @solidity memory-safe-assembly
+        assembly {
+            // Emit the {Withdrawal} event.
+            mstore(0x00, amount)
+            log2(0x00, 0x20, _WITHDRAWAL_EVENT_SIGNATURE, to)
```
Comment (Vectorized) [inline src/tokens/WETH.sol:118]: Upper 96 bits of `to` must be cleaned for events.

File: src/tokens/WETH.sol [hunk 3]
```diff
@@ -84,6 +84,48 @@ contract WETH is ERC20 {
...
+    /// @dev Deposits `amount` ETH of the caller and mints `amount` WETH to the `to`.
+    ///
+    /// Emits a {Deposit} event.
+    function depositTo(address to) public payable virtual {
+        _mint(to, msg.value);
+        /// @solidity memory-safe-assembly
+        assembly {
+            // Emit the {Deposit} event.
+            mstore(0x00, callvalue())
+            log2(0x00, 0x20, _DEPOSIT_EVENT_SIGNATURE, to)
```
Comment (Vectorized) [inline src/tokens/WETH.sol:96]: Upper 96 bits of `to` must be cleaned for events.

File: src/tokens/WETH.sol (hunk 2 above)
Comment (Atarpara) [reply src/tokens/WETH.sol:118]: not need this solc complier handle this for up. 
Try this: https://twitter.com/ATarpara/status/1665955030455980034

File: src/tokens/WETH.sol (hunk 3 above)
Comment (Atarpara) [reply src/tokens/WETH.sol:96]: not needed this solc complier handle this for up. 
Try this: https://twitter.com/ATarpara/status/1665955030455980034

File: src/tokens/WETH.sol (hunk 1 above)
Comment (Atarpara) [reply src/tokens/WETH.sol:107]: Done

**Comment by Vectorized** [comment]
I'll merge to `cweth2` first.

**Review by Vectorized — APPROVED** [review]

**Pull request merged**